app/boards/shields/zmk_uno/boards/nrf52840dk_nrf52840.overlay

```diff
@@ -20,5 +29,38 @@ encoder: &qdec0 {
     led-pre = <0>;
     steps = <80>;
     pinctrl-0 = <&qdec_default>;
-    pinctrl-names = "default";
+    pinctrl-1 = <&qdec_sleep>;
+    pinctrl-names = "default", "sleep";
 };
+
+/ {
+    behaviors {
+        soft_off: soft_off {
+            compatible = "zmk,behavior-soft-off";
+            #binding-cells = <0>;
+            status = "okay";
+        };
+    };
+
+    wakeup_source: wakeup_source {
+        compatible = "zmk,wakeup-trigger-key";
+        status = "okay";
+
+        trigger = <&button0>;
+        wakeup-source;
+    };
+
+    soft_off_wakers {
+        compatible = "zmk,soft-off-wakeup-sources";
+        status = "okay";
+
+        wakeup-sources = <&wakeup_source>;
+    };
+
+    soft_off_behavior_key {
+        compatible = "zmk,behavior-key";
```

Comment:
Now that I see what this does, I think this naming could be very confusing. It fits the pattern of `zmk,behavior-*` for things that can be bound to a keymap, but it is fundamentally a different thing. It's more akin to a combination of kscan driver and keymap than a behavior.

We could also make it even clearer that this isn't a traditional behavior by avoiding the `zmk,behavior-*` pattern and/or using `gpio-key` instead of `key`. How about something like `zmk,gpio-key-behavior-trigger`?

Reply:
*grumble grumble* Yes, you're absolutely right. Renamed for clarity. Thanks!